addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -210,7 +210,7 @@ export class DiscussCoreCommon {
                 ...channelData,
             });
         }
-        if (!channel.is_pinned) {
+        if (!channel.is_pinned && !channel.readonly) {
```

Comment:
Having the field named readonly is really confusing if it means "is member".
I'd suggest a rename to be close to the truth.